src/library_webgl.js

```diff
@@ -1220,14 +1214,16 @@ for (/**@suppress{duplicate}*/var i = 0; i < {{{ GL_POOL_TEMP_BUFFERS_SIZE }}};
     },
 #endif
 
-    getExtensions() {
-      // .getSupportedExtensions() can return null if context is lost, so coerce to empty array.
-      var exts = GLctx.getSupportedExtensions() || [];
+  },
+
+  $glGetExtensions__internal: true,
+  $glGetExtensions__deps: ['$getEmscriptenSupportedExtensions'],
+  $glGetExtensions() {
```

Comment:
nit: This naming convention makes glGetExtensions sound like a gl.h entrypoint (except for the `$`), but it isn't.
Hopefully it can be somewhat clear how it's different from getEmscriptenSupportedExtensions. Right now it feels arbitrary which ones is called in which place.

(I'm actually not totally sure why we expose GL_EXTENSIONS entries without prefix - I think OpenGL always prefixes them, and only WebGL, which doesn't even have `gl.getString(gl.EXTENSIONS)`, doesn't. But that's beside the point.)

Maybe
getEmscriptenSupportedWebGLExtensionsUnprefixed
getEmscriptenSupportedWebGLExtensionsPlusPrefixed

("WebGL" distinguishing it from other GL extensions like the ones in glemu.)

Reply:
> I'm actually not totally sure why we expose GL_EXTENSIONS entries without prefix

The reason for this was:

1. originally from dawn of time, Emscripten exposed GL_EXTENSIONS list without GL_ prefixing the extensions,
2. when working on porting Unreal Engine 3 in 2013, Vlad Vukicevic realized that of course the extensions won't match what UE3 would expect, so he asked me to add GL_EXTENSIONS to advertise the prefixed variants,
3. we didn't dare to remove advertising the original unprefixed variants, at the basis of "maybe that could break existing codebases targeting Emscripten that might be reading the unprefixed extensions".

So we decided to leave them in both forms.

Reply:
> nit: This naming convention makes glGetExtensions sound like a gl.h entrypoint (except for the $), but it isn't.

Agree with this. With respect to the GL spec, naming custom functions with `glFoo()` is a bit similar to adding new functions in the `std::` namespace in C++.

Not that we'd ever have problems with a function named `glGetExtensions`, though there is some sense of misplacedness to it.

Maybe this could be `$webglGetExtensions` instead?

Reply:
Done